test/RSocketClientTest.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "RSocketTests.h"
+
+#include <folly/io/async/ScopedEventBaseThread.h>
+#include <gtest/gtest.h>
+
+#include "rsocket/transports/tcp/TcpConnectionFactory.h"
+
+using namespace rsocket;
+using namespace rsocket::tests;
+using namespace rsocket::tests::client_server;
+
+TEST(RSocketClient, ConnectFails) {
+  folly::ScopedEventBaseThread worker;
+
+  folly::SocketAddress address;
+  address.setFromHostPort("localhost", 9876);
```

Comment:
Nit: I think port==1 is better for an "obvious can't connect failure" type situation.  Wasn't immediately obvious that 9876 was expected to have nothing bound to it.